### PR TITLE
don't call SizeOf for type parameters

### PR DESCRIPTION
**Pull request by mcdoker18**
StdSizes.SizeOf method doesn't support type parameters. https://github.com/golang/go/blob/78b722d8c2f764c3048c6f0344e9ebcd2687813d/src/go/types/sizes.go#L156-L159

Fixes go-critic/go-critic#1223

### CODE REVIEW

**Review by cristaloleg — DISMISSED** [review]
SGTM

**Comment by mcdoker18** [comment]
```
hugeParam_checker.go:53:28: TypeParam not declared by package types
```

@cristaloleg  I use go1.18 locally, so I didn't mention that go-critic still uses 1.17.

Do you have any plan\issue to upgrade go-critic to 1.18? I can try in separate pull request

**Comment by cristaloleg** [comment]
Well, that's a hard question to be honest. We should bump go-critic to Go 1.18 somewhere in July. CC: @quasilyte

File: checkers/hugeParam_checker.go [hunk 1]
```diff
@@ -49,6 +50,9 @@ func (c *hugeParamChecker) checkParams(params []*ast.Field) {
 	for _, p := range params {
 		for _, id := range p.Names {
 			typ := c.ctx.TypeOf(id)
+			if _, ok := typ.(*types.TypeParam); ok {
```
Comment (quasilyte) [inline checkers/hugeParam_checker.go:53]: Shouldn't we use a polyfill like `x/exp/typeparams` to make this code compileable with older Go versions?
golangci-lint can't deprecate pre-generics release yet AFAIK.

Comment (quasilyte) [reply checkers/hugeParam_checker.go:53]: https://pkg.go.dev/golang.org/x/exp/typeparams

Comment (mcdoker18) [reply checkers/hugeParam_checker.go]: Fixed.

If we support go 1.17 then I should hide the new test cases using build constraints.

Comment (mcdoker18) [reply checkers/hugeParam_checker.go]: BTW, If you'd like I can add go1.18 CI support in a separated PR.

https://github.com/mcdoker18/go-critic/pull/3/commits/2041b982633b256b73c61521614172b11133f1dd

**Comment by quasilyte** [comment]
The code looks good now.

**Comment by mcdoker18** [comment]
@cristaloleg rebased with master

**Review by quasilyte — APPROVED** [review]

**Comment by mcdoker18** [comment]
@cristaloleg Hi, can you take a look?

**Pull request merged**